### DISCUSSION TITLE
Any ideas how to get Steam notifications to behave well with hyprland + waybar?

### DISCUSSION THREAD

**Discussion by Hubro** [hyprwm/Hyprland#3090] [Q&A]
Waybar needs to have `"layer": "top"` set, otherwise context menus from the system tray will appear behind windows.

Unfortunately, Steam rolls its own notification windows and don't care about the "reserved area" of the screen. The notifications show up in the very bottom right corner of the screen, partly underneath my waybar, which means I can only see half of the notification.

Does anybody have any tips for an easy way to resolve this?

I have several ideas:

- Add a rule to Steam notifications that moves them to the "top" or "overlay" layers. (Is this even possible? I couldn't find a rule like that in the wiki.)
- Add a rule that moves Steam notifications up by the height of waybar
- (Even cooler) Add a script that runs whenever any new window opens and (if necessary) moves the window enough to be within the available area

I don't know which of these would be the least effort, and I also don't really want to spend a bunch of time duplicating something somebody else has already done, so I figured this was a good place to ask.

**Comment by memchr** [comment]
How do I manually trigger a notification from steam?

**Reply by Hubro** [reply]
No idea, I think you just have to get a friend to assist you by spamming you on demand :smile:

Alternatively you could create a second Steam account and log in on your phone to send yourself messages.

**Comment by DeveloperMikey** [comment]
Has anyone figured out how to move the notification up with a windowrule? The title of the notifications are always "notificationtoasts_*_desktop" with the star being an incremental number for every notification.

**Comment by VeilSilence** [comment]
Also looking for solution. With dual monitors, steam notification appear at wrong monitor.
Tried `windowrulev2 = monitor DP-1,class:^(steam),title:^(.*notificationtoasts*.)(.*)` for example, and it didn't work.
Really wish there was a setting like in KDE for "main" monitor.

**Reply by zakk4223** [reply]
Probably the 'no xwayland primary' issue. Either write a convoluted script to do xrandr --output <main monitor> --primary or use my plugin: https://github.com/zakk4223/hyprXPrimary

**Reply by VeilSilence** [reply]
Using plugin already. Not helping for me. I'm using steam on start so, probably because of that.
Fixed with `exec-once = xrandr --output DP-1 --primary && steam-runtime &`

**Reply by nediaH** [reply]
This is over a year old but I just wanted to thank you for fixing this infuriating issue for me.
